backend/src/services/cars/cars-local-search.service.ts

```diff
@@ -2,6 +2,24 @@ import { CarsSearchParams, SearchResult } from '@autoline/shared';
 import { prisma } from '@data/prisma-client';
 
 const carsSearch = async (data: CarsSearchParams): Promise<SearchResult[]> => {
+  const brandDetails = data.brandDetails;
+
+  const brandFiltersWhere = brandDetails && {
+    OR: [
+      ...brandDetails.map(({ modelIds, brandId }) => ({
+        ...(modelIds &&
+          modelIds.length && {
+            id: {
+              in: modelIds,
+            },
+          }),
+        brand: {
+          id: brandId,
+        },
+      })),
+    ],
```

Comment:
You already create a new array with the `.map`. Why to you spread it again into an empty array?